daffodil-core/src/main/scala/org/apache/daffodil/dsom/SchemaDocument.scala

```diff
@@ -124,6 +125,46 @@ final class XMLSchemaDocument(xmlArg: Node,
     res
   }.value
 
+  /**
+   *  True if root xs:schema element has an xmlns that uses the DFDL URI.
+   *
+   *  This could be a prefix definition (Most likely xmlns:dfdl='...' but could
+   *  be some other prefix.)
+   *
+   *  Or very very unlikely, it could be the default namespace.
+   */
+  private def hasDFDLNamespaceDefinition: Boolean = {
+    val scope = xml.scope
+    val pre = scope.getPrefix(XMLUtils.DFDL_NAMESPACE)
+    val hasSomePrefixForDFDLNamespace = pre ne null
+    val res = hasSomePrefixForDFDLNamespace ||
+      {
+        val default = scope.getURI(null)
+        val hasDefaultNamespaceAsDFDLNamespace = default == XMLUtils.DFDL_NAMESPACE.toString
+        hasDefaultNamespaceAsDFDLNamespace
+      }
```

Comment:
Could the inside of the brackets be moved to before the res so it just becomes ``val res = hasSomePrefixForDFDLNamespace || hasDefaultNamespaceAsDFDLNamespace``? This currently avoids some code if DFDL is the not the default namespace, but that doesn't seem all that intensive and this is a bit hard to read.

Reply:
Yes, another way to do this is
{code}
val = cheapThingToCompute = .....
lazy val = expensiveThingToCompute = ....
val res = cheapThingToCompute || expensiveThingToCompute
{code}
This gets the clarity and the efficiency minus the lazy val overhead. I wouldn't do this is the runtime though, but in the compiler I'm ok with it.